components/SplitPane/library/src/desktopMain/kotlin/org/jetbrains/compose/splitpane/DesktopSplitPane.kt

```diff
@@ -51,101 +50,91 @@ internal actual fun SplitPane(
             val secondMinSizePx = secondPlaceableMinimalSize.value * density
 
             with(splitPaneState) {
-                val constrainedMin = constraints.minByDirection(isHorizontal) + firstMinSizePx
-                val constrainedMax =
-                    (constraints.maxByDirection(isHorizontal).toFloat() - secondMinSizePx).let {
-                        if (it <= 0 || it <= constrainedMin) {
-                            constraints.maxByDirection(isHorizontal).toFloat()
-                        } else {
-                            it
-                        }
-                    }
+                val firstMeasurable = measurables[0]
+                val splitterMeasurable = measurables[1]
+                val secondMeasurable = measurables[2]
+                val handleMeasurable = measurables[3]
 
-                if (minPosition != constrainedMin) {
-                    maxPosition = constrainedMin
-                }
+                // Need the size of the splitter to determine the min/max position
+                // Constrain the splitter only on the "other" axis
+                val splitterConstraints =
+                    if (isHorizontal) {
+                        Constraints(
+                            minHeight = constraints.minHeight,
+                            maxHeight = constraints.maxHeight
+                        )
```

Comment:
I guess it's a good candidate to extract this to extension function for readability